src/main/webapp/app/text-assessment/text-assessment.component.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, OnInit } from '@angular/core';
+import { TextExercise } from 'app/entities/text-exercise';
+import { TextSubmission } from 'app/entities/text-submission';
+import { HighlightColors } from '../text-shared/highlight-colors';
+import { JhiAlertService } from 'ng-jhipster';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Result, ResultService } from 'app/entities/result';
+import { TextAssessmentsService } from 'app/entities/text-assessments/text-assessments.service';
+import { Feedback } from 'app/entities/feedback';
+import { Participation } from 'app/entities/participation';
+
+@Component({
+    providers: [TextAssessmentsService],
+    templateUrl: './text-assessment.component.html',
+    styles: []
+})
+export class TextAssessmentComponent implements OnInit {
+    text: string;
+    participation: Participation;
+    submission: TextSubmission;
+    result: Result;
+    assessments: Feedback[] = [];
+    exercise: TextExercise;
+    totalScore = 0;
+    assessmentsAreValid: boolean;
+    invalidError: string;
+    isAuthorized = true;
+    accountId = 0;
+    busy = true;
+
+    public getColorForIndex = HighlightColors.forIndex;
+
+    constructor(
+        private jhiAlertService: JhiAlertService,
+        private modalService: NgbModal,
+        private router: Router,
+        private route: ActivatedRoute,
+        private resultService: ResultService,
+        private assessmentsService: TextAssessmentsService
+    ) {
+        this.assessments = [];
+        this.assessmentsAreValid = false;
+    }
+
+    public ngOnInit(): void {
+        this.busy = true;
+        const exerciseId = Number(this.route.snapshot.paramMap.get('exerciseId'));
+        const submissionId = Number(this.route.snapshot.paramMap.get('submissionId'));
+
+        this.assessmentsService.getFeedbackDataForExerciseSubmission(exerciseId, submissionId).subscribe(participation => {
+            this.participation = participation;
+            this.submission = <TextSubmission>this.participation.submissions[0];
+            this.exercise = <TextExercise>this.participation.exercise;
+            this.result = this.participation.results[0];
+            this.assessments = this.result.feedbacks;
+            if (!this.assessments) {
+                this.assessments = [];
+            }
```

Comment:
You can simplify the previous 4 lines with 

```javascript
this.assessments = this.result.feedbacks || [];
```

Reply:
:+1: